pkg/apis/pipeline/v1alpha1/taskrun_types.go

```diff
@@ -42,6 +42,8 @@ type TaskRunSpec struct {
 	Results Results `json:"results"`
 	// +optional
 	Generation int64 `json:"generation,omitempty"`
+	// +optional
+	ServiceAccount string `json:"serviceAccount"`
```

Comment:
i'm probably just getting myself confused, but im surprised that `TaskRun` doesn't have a reference to `PipelineParams(es)` already - if it did, then we could get the `ServiceAccount` out of there, instead of adding another field?

On the one hand it's nicer to be able to use a `TaskRun` without `PipelineParams` but on the other hand we probably want the log and result endpoints for the `TaskRun` anyway

Reply:
We need ServiceAccount on taskrun so we can run them without a pipeline, but also the pipelinerun controller will set this field from the value in pipeline params

Reply:
>  but im surprised that TaskRun doesn't have a reference to PipelineParams(es) already 

Task and Taskrun should not reference to pipeline or pipeline params objects. It creates cyclic references which is bad

Reply:
> Task and Taskrun should not reference to pipeline or pipeline params objects

Pipeline params won't reference anything, so there's no cycle there.

In that case, what are your thoughts about how we handle uploading results & logs for TaskRuns when they are run manually?

Reply:
Results are declared on `taskrun.spec` so user can provide these variables if Taskrun is run without pipeline context.

Reply:
If results need service account to upload logs then `taskrun.spec.serviceAccountName` is used   to achieve that.

Reply:
> Results are declared on taskrun.spec so user can provide these variables if Taskrun is run without pipeline context.

kk, we can go forward with that for now then and see how it goes!